src/metabase/api/search.clj

```diff
@@ -212,17 +177,23 @@
                [:lower column]
                (wildcard-match token)])))))
 
+(defn- build-optional-filters
```

Comment:
I think this function should live in `search.filter`, and also include the `archived-where-clause`, becoming `build-filters`. That way `archived?` is just another filter and the details of which filters are applied feel more hidden from this namespace. The reader of `base-where-clause-for-model` should only need to know that all the filters (including `archived?`) are applied appropriately.

Reply:
done. created a new `build-filters` in `search.filter` that build both archived and optional filters